packages/host/app/components/operator-mode/create-file-modal.gts

```diff
@@ -87,4 +153,58 @@ export default class CreateFileModal extends Component<Signature> {
   @action removeSelectedRealm() {
     this.selectedRealmURL = undefined;
   }
+
+  private chooseCardInstanceType = restartableTask(async () => {
+    this.selectedCatalogEntry = await chooseCard({
+      filter: {
+        on: catalogEntryRef,
+        eq: { isField: false },
+      },
+    });
+  });
+
+  private createCardInstance = restartableTask(async () => {
+    if (!this.selectedCatalogEntry?.ref || !this.selectedRealmURL) {
+      return;
+    }
+
+    let { ref } = this.selectedCatalogEntry;
+
+    let doc: LooseSingleCardDocument = {
+      data: {
+        meta: {
+          adoptsFrom: ref,
+          realmURL: this.selectedRealmURL.href,
+        },
+      },
+    };
```

Comment:
is there a reason that we make the document before we fully resolve the ref? do we know that changing the `ref` after making the doc will update the doc?

Reply:
I moved this to after the ref is converted to absolute url but it is still fully resolved elsewhere. We've been using this pattern to create cards. If it is incorrect, we should change it in other places too